bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BArrayType.java

```diff
@@ -128,8 +128,23 @@ public boolean equals(Object obj) {
 
     @Override
     public String toString() {
-        StringBuilder sb = new StringBuilder(elementType.toString());
-        return size != -1 ? sb.append("[").append(size).append("]").toString() : sb.append("[]").toString();
+        StringBuilder sb = new StringBuilder();
+        BType element = elementType;
+        sb.append(getSizeString());
+        while (true) {
+            if (!(element instanceof BArrayType)) {
```

Comment:
why is this check moved outside of `while` loop's condition?

Reply:
Both methods works fine. But I used this as I thought this code is more clear to look at.

Reply:
```java
while (true) {
    if (!(element instanceof BArrayType)) {
        break;
     }
    ...
}
```

vs 

```java
while (element instanceof BArrayType) {
    ...
}
```

I think the second one is much clearer.

Reply:
changed it in latest commit